src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -35,6 +36,11 @@ internal class InMemoryPartitionedAppTokenCacheAccessor : ITokenCacheAccessor
         protected readonly ICoreLogger _logger;
         private readonly CacheOptions _tokenCacheAccessorOptions;
 
+        // Approximate size of cache item objects
+        private const long AccessTokenSizeInBytes = 4500;
```

Comment:
My main concern is that a constant approximation will not work well: 

1. AAD allows for configuration of optional claims
2. POP tokens are a bit bigger because they contain the POP key id
3. AAD is adding support for SAML tokens (there is already an OBO flow for this). We won't be able to update this aproximation for all token types.

I think there are a few ways out of this: 

1. Allow customers to define this constant (this is what SAL does). But this is a pretty "obscure" scenario
2. Use non-constant size approximation, i.e. we know the size of the actual token (it's a string) at which we add some metadata (fairly constant)
3. Do not expose sizeLimit, expose countLimit

Reply:
@dannybtsai @henrik-me - can you please review this strategy?

Reply:
Is there is reason to use a constant value instead of the actual size (the #2 option)?

Reply:
@bgavrilMS 
1. Hmm, interesting, I'd like to see the implementation in SAL. How would customers know that constant?
2. Yea, metadata should be fairly constant. This does seem like a better approximation.
3. Specific to the user cache, what would the count represent: only access tokens, all tokens? Because in both cases, the count would not represent the whole cache size accurately.

@dannybtsai 
How would you measure the actual size of objects in memory? We could measure the strings size in cache objects like Bogdan mentioned below and in Microsoft.Identity.Web we serialize the whole cache, so take size that way. But in this case cache entry is an object.